test/VoteDelegate.t.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+import "dss-test/DssTest.sol";
+
+import {VoteDelegate, ChiefLike, PollingLike, GemLike} from "src/VoteDelegate.sol";
+
+interface ChiefExtendedLike is ChiefLike {
+    function approvals(address) external view returns (uint256);
+}
+
+interface GemLikeExtended is GemLike {
+    function allowance(address, address) external view returns (uint256);
+    function balanceOf(address) external view returns (uint256);
+    function mint(address, uint256) external;
+}
+
+contract VoteDelegateTest is DssTest {
+    address constant c1 = address(0x1);
+    address constant c2 = address(0x2);
+
+    VoteDelegate proxy;
+    GemLikeExtended gov;
+    ChiefExtendedLike chief;
+    PollingLike polling;
+
+    address delegate = address(111);
+    address delegator1 = address(222);
+    address delegator2 = address(333);
+
+    event Lock(address indexed usr, uint256 wad);
+    event Free(address indexed usr, uint256 wad);
```

Comment:
In case you take the previous two suggestions

```suggestion
    event Free(address indexed usr, uint256 wad);
    event Voted(address indexed voter, uint256 indexed pollId, uint256 indexed optionId);
```